src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -125,6 +125,7 @@ public abstract class PeerConnector : IPeerConnector
             this.nodeLifetime = nodeLifetime;
             this.NodeSettings = nodeSettings;
             this.peerAddressManager = peerAddressManager;
+            this.Requirements = new NetworkPeerRequirement { MinVersion = this.NodeSettings.ProtocolVersion };
```

Comment:
If you set min version here why not also set the peer service to nothing and only override it in discovery?

Reply:
Hi @dangershony! The reason here is that MinVersion is the same for all connectors but the requirements (can) differ. I like to code in such a way that if a property or field of ALL classes is not exactly the same, their definition or initialization should be in the respective sub classes. That way explicit as to where it is set. Thoughts?

Reply:
completely different thought - was this minimal version mechanism always there, set to this NodeSettings.ProtocolVersion? 

and second question - would such a setting not disqualify older nodes to communicate with this node?

Reply:
@Aprogiena before PeerConnectors we had NodeGroups, which used to set this value on startup. With regards to your second question it looks like we are checking this value in NetworkPeerRequirement and subsequently in VersionHandshakeAsync where if its not valid, we disconnect the peer.

Hope this answers your question 👍

Reply:
this is not for this PR, just thinking out load

so yeah, the important thing here is what is the default value and i can see the default value for this is 

```
        public const ProtocolVersion SupportedProtocolVersion = ProtocolVersion.SENDHEADERS_VERSION;
```

which is quite a big version to be honest, it would really disqualify older nodes to be connected, which is weird because if we look here  https://chainz.cryptoid.info/strat/#!network we can see that the majority of the network is not on that version (0x70012), most nodes are using 0x70000 and unless i'm missing something, those should be disconnected, but that does not really match what we see during tests, so i'm probably missing something